### PR TITLE
Use Filesystem label in container_runtime_extra_storage

### PR DESCRIPTION
**Pull request by maci0**
Use Filesystem label in container_runtime_extra_storage instead of device names.
Also skips if the mountpoint already exists.
This deals with changing device order for example in aws.
/dev/nvme1n1 may be /dev/nvme2n1 on another boot.
Running the  prerequisties playbook could potentially mess up the mountpoints if that happens.

This makes the `label` option mandatory in the disk description in the inventory file.
Example:
```
# Storage config
container_runtime_extra_storage='[{"device":"/dev/nvme1n1","path":"/var/lib/origin/openshift.local.volumes","filesystem":"xfs","options":"gquota", "label": "localvol"},{"device":"/dev/nvme2n1","path":"/var/lib/containers","filesystem":"xfs","options":"gquota", "label": "containers"}]'

[masters:vars]
container_runtime_extra_storage='[{"device":"/dev/nvme2n1","path":"/var/lib/origin/openshift.local.volumes","filesystem":"xfs","options":"gquota", "label": "localvol"},{"device":"/dev/nvme1n1","path":"/var/lib/containers","filesystem":"xfs","options":"gquota", "label": "containers"},{"device":"/dev/nvme3n1","path":"/var/lib/etcd","filesystem":"xfs","options":"gquota", "label": "etcd"}]'
```

NOTICE
======

Master branch is closed! A major refactor is ongoing in devel-40.
Changes for 3.x should be made directly to the latest release branch they're
relevant to and backported from there.

### CODE REVIEW

**Comment by maci0** [comment]
right now it will fail if label is not define. may need to add a check to make sure label is defined and <= 12 characters

File: roles/container_runtime/tasks/extra_storage_setup.yml [hunk 1]
```diff
@@ -4,15 +4,22 @@
     fstype: "{{ item.filesystem }}"
     dev: "{{ item.device }}"
     force: "{{ item.force|default(omit) }}"
-    opts: "{{ item.mkfs_opts|default(omit) }}"
+    opts: "{{ item.mkfs_opts|default(None) }} -L {{ item.label }}"
   with_items: "{{ container_runtime_extra_storage }}"
+  when:
+    - 'container_runtime_extra_storage is defined'
```
[the remark this answers is not in the document]
Comment (maci0) [reply roles/container_runtime/tasks/extra_storage_setup.yml]: true, had this in there for more test and was using a different var name earlier. it doesn't hurt though.

File: roles/container_runtime/tasks/extra_storage_setup.yml [hunk 2]
```diff
@@ -4,15 +4,22 @@
     fstype: "{{ item.filesystem }}"
     dev: "{{ item.device }}"
     force: "{{ item.force|default(omit) }}"
-    opts: "{{ item.mkfs_opts|default(omit) }}"
+    opts: "{{ item.mkfs_opts|default(None) }} -L {{ item.label }}"
   with_items: "{{ container_runtime_extra_storage }}"
+  when:
+    - 'item.label|length<=12'
+    - 'item.path not in "{{ ansible_mounts }}"'
```
Comment (vrutkovs) [inline roles/container_runtime/tasks/extra_storage_setup.yml]:
```suggestion
    - 'item.path not in ansible_mounts"'
```

```
Error: Usage of Jinja2 templating delimiters in when conditions is deprecated in Ansible 2.3.
  File: /home/travis/build/openshift/openshift-ansible/roles/container_runtime/tasks/extra_storage_setup.yml
  Found: "item.path not in "{{ ansible_mounts }}""
  Found: "item.path not in "{{ ansible_mounts }}""
```

**Comment by vrutkovs** [comment]
Could you change the branch from `master` to `release-3.11`? master branch would soon be reworked

**Comment by mtnbikenc** [comment]
Please reopen this pull request after addressing the review comments and rebase your branch against release-3.11.

**Pull request closed without merging**